### ISSUE TITLE
Feature request - Mac OS notifications per job status

### ISSUE THREAD

**Issue by amaisano** [rondorn/CiPol#5]
It would be amazing if a job going from "in-progress" to "complete" or "idle" (and other combos) would trigger a Mac OS notification. Something like:

```
 ----------------------------------
|  Job sw/D10/qal now IDLE (PASS)  |
 ----------------------------------
```

All we get right now is if _any_ build failed. Not the status of a particular job.

**Comment by amaisano** [comment]
Right-clicking a job and having the option to both "Subscribe for Notifications" and name the notification would be good too, Then you can opt-in and opt-out of jobs on the fly and have better labels for each.

**Comment by rondorn** [comment]
On a trip out of the country right now, but that’s a good suggestion. I’ll look into that.Thanks for the suggestion.Sent from my iPhoneOn Jan 30, 2024, at 1:46 PM, Adamo Maisano ***@***.***> wrote:﻿
Right-clicking a job and having the option to both "Subscribe for Notifications" and name the notification would be good too, Then you can opt-in and opt-out of jobs on the fly and have better labels for each.

**Comment by rondorn** [comment]
So I am a little confused by what you are asking for now that I have time to sit and read your message and look at the program

- So with the "Alerting" check box, you can continue to monitor a job, but turn off alerts for that job. This seems to function exactly like the Subscribe you are talking about.
- The Job does get a name that appears in the alert, though that is automatically taken from the Jenkins job. Do you want the option to custom name the job outside of Jenkins?
- The program does alert whenever a job changes state, and the alert does give the name of the job and the state that it is now. Are you asking to include the value of the previous state?

Thanks

**Comment by amaisano** [comment]
Perhaps there is a bug then - I receive no notifications when a job changes status. I expected one when job X went from `In Progress` to `Idle` but got nothing.

I only received a notification once when a build went from failing to passing.

**Comment by rondorn** [comment]
Ok, I get it. Right now I only alert on the Pass/Fail state. Pass, Fail, and Unknown.

You want an optional alert on the "Status" field? So you know when the job is completed, regardless of the status?

**Comment by amaisano** [comment]
Exactly - our team does multiple builds on multiple environments each day, and it would be amazing to get pinged when something is "done building" without having to dig into the website every time.

**Comment by rondorn** [comment]
Let me see what I can do.I have a lot on my plate right now, but I will put this on the list to get to.Thanks for the feedbackSent from my iPadOn Feb 6, 2024, at 6:52 AM, Adamo Maisano ***@***.***> wrote:﻿
Exactly - our team does multiple builds on multiple environments each day, and it would be amazing to get pinged when something is "done building" without having to dig into the website every time.

**Comment by rondorn** [comment]
I am adding a new feature that is similar to what you are asking for. It is in review at Apple now.If a job is failing and the job is run again, and continue to fail there will be an optional alert that will go off. Right now it will not work if the job is passing and continue to pass though.This is addition to notifying when the job changes state (fail to pass, or pass to fail)Check it out and let me know if this works for you.Sorry for the long delay in updating the app.Sent from my iPadOn Jan 30, 2024, at 9:51 AM, Adamo Maisano ***@***.***> wrote:﻿
It would be amazing if a job going from "in-progress" to "complete" or "idle" (and other combos) would trigger a Mac OS notification. Something like:
 ----------------------------------
|  Job sw/D10/qal complete (Pass)  |
 ----------------------------------

All we get right now is if any build failed. Not the status of a particular job.
